### ISSUE TITLE
Child association documents are duplicated when inserted from custom controller action (using mongoDB)

### ISSUE THREAD

**Issue by dinana** [balderdashy/sails-mongo#208]
Steps to reproduce:
1. Clone this repo: https://github.com/dinana/testDup
2. Create User
3. Create post by POST'ing to localhost:1337/user/<created_user_id>/postsCreated
4. Create comments by POST'ing to localhost:1337/post/<created_post_id>/comments

Now comments are created in their own collection, normal.

5. Try posting to localhost:1337/post/addcomments with a postid parameter equivalent to the created post id in 3.

Now the post document in the posts collection has an embedded array with a subset of the comments (usually all comments created before 5), this is a duplication from the comments created in the comments collection whose post attribute is set to the created post id in 3.

**Comment by dinana** [comment]
I'm surprised this haven't been solved till now, anyways, here's the workaround I've used: Having the last example in mind, you just create a new comment, add it to the comments collection and have its post field set to its parent post.

**Comment by sailsbot** [comment]
Thanks for posting, @dinana.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
